.github/workflows/e2e-tests-main-devnet.yml

```diff
@@ -532,6 +532,103 @@ jobs:
 
+  check-runtime-change:
+    name: Inspect whether runtime version has been changed (comparing to main)
+    runs-on: ubuntu-latest
+    outputs:
+      runtime-updated: ${{ steps.inspect.outputs.diff }}
+    steps:
+      - name: Checkout current and old source code
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: Inspect versions
+        id: inspect
+        run: |
+          if ! git diff HEAD origin/main -- bin/runtime/src/lib.rs | grep -q ' spec_version: '
+          then
+            echo "::set-output name=diff::1"
+          else
+            echo "::set-output name=diff::0"            
+          fi
+
+  build-new-runtime-and-try_runtime:
+    name: Build new runtime and try_runtime tool
+    needs: [ check-runtime-change ]
+    if: ${{ needs.check-runtime-change.outputs.runtime-updated != 0 }}
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout source code
+        uses: actions/checkout@v3
+
+      - name: Install Rust toolchain
+        uses: actions-rs/toolchain@v1
+
+      - name: Install WASM target
+        run: rustup target add wasm32-unknown-unknown
+
+      - name: Restore cache
+        uses: ./.github/actions/restore-cache
+        with:
+          target-key: try-runtime
+          cargo-key: new
+
+      - name: Build
+        run: cargo build --release -p aleph-node --features "try-runtime short_session"
+
+      - name: Upload new runtime
+        uses: actions/upload-artifact@v3
+        with:
+          name: aleph-runtime-try_runtime
+          path: target/release/wbuild/aleph-runtime/aleph_runtime.compact.wasm
+          if-no-files-found: error
+          retention-days: 7
+
+      - name: Upload try-runtime
+        uses: actions/upload-artifact@v3
+        with:
+          name: try_runtime
+          path: target/release/aleph-node
+          if-no-files-found: error
+          retention-days: 7
+
+      - name: Cleanup cache
+        uses: ./.github/actions/post-cache
+
+  test-runtime-update:
+    name: Test runtime update with try-runtime tool
+    runs-on: ubuntu-latest
+    needs: [ build-new-runtime-and-try_runtime ]
+    steps:
+      - name: Checkout source code
+        uses: actions/checkout@v3
+
+      - name: Download all artifacts
+        uses: actions/download-artifact@v3
+        with:
+          path: local-tests/
+
+      - name: Setup Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.8'
+
+      - name: Test update
+        timeout-minutes: 20
+        env:
+          NEW_RUNTIME: aleph-runtime-try_runtime/aleph_runtime.compact.wasm
+          TRY_RUNTIME: try_runtime/aleph-node
+        working-directory: local-tests/
+        run: |
+          chmod +x ${{ env.TRY_RUNTIME }}
+          pip install -r requirements.txt
+          ./test_runtime_update.py \
+              wss://ws.test.azero.dev:443 \
+              ../bin/node/src/resources/testnet_chainspec.json \
```

Comment:
We need to use _new_ runtime, ie generated locally from what's on PR. Hence we need to generate chain spec from new node version.

Reply:
My bad, I did not read review carefully, all good.